### PR TITLE
Added support for reading Parquet V2 pages

### LINKED ISSUE #5429: Parquet Errors Have Too Little Context
As an adminstrator, I want the errors from reading Parquet files to include the file name, so that it is easier to understand why things are broken.

This error being a Requirement failure, without providing any context for which file is broken makes it very difficult to understand the root cause of errors.
image

### PR DESCRIPTION
**Pull request by malhotrashivam**
Closes #5429

### CODE REVIEW

File: extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java [hunk 1]
```diff
@@ -210,59 +222,100 @@ private IntBuffer readKeysFromDataPage(IntBuffer keyDest, int nullPlaceholder, S
...
+            final RunLengthBitPackingHybridBufferDecoder dlDecoder =
+                    new RunLengthBitPackingHybridBufferDecoder(path.getMaxDefinitionLevel(),
+                            pageBytes.slice().limit(length));
+            pageBytes.position(pageBytes.position() + length);
+            return dlDecoder;
+        }
+        return null;
+    }
+
+
```
Comment (rcaudy) [inline extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java:309]:
```suggestion
```

File: extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java [hunk 2]
```diff
@@ -210,59 +222,100 @@ private IntBuffer readKeysFromDataPage(IntBuffer keyDest, int nullPlaceholder, S
...
+                final ByteBuffer bytes = page.getBytes().toByteBuffer();
                 bytes.order(ByteOrder.LITTLE_ENDIAN);
-                int length = bytes.getInt();
+                final int length = bytes.getInt();
                 return readRepetitionLevels(bytes.slice().limit(length));
             } else {
                 return page.getValueCount();
             }
-        } catch (IOException e) {
+        } catch (final IOException e) {
```
Comment (rcaudy) [inline extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java:268]: It seems to me like:
1. The formatting might leave something to be desired.
2. We're inconsistent about catching this sort of thing. For instance, we don't do so on the key-reading path, at least not where I would expect.

File: extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java [hunk 3]
```diff
@@ -323,89 +377,104 @@ private int readRepetitionLevels(ByteBuffer byteBuffer) throws IOException {
...
+    // TODO Create an issue for testing and optimizing further
+    @SuppressWarnings("unused")
+    private void readKeysWithNulls(
+            final IntBuffer keysBuffer,
+            final int nullPlaceholder,
+            final RunLengthBitPackingHybridBufferDecoder dlDecoder,
+            final ValuesReader dataReader) throws IOException {
+        if (path.getMaxRepetitionLevel() > 0) {
+            throw new UnsupportedOperationException("Repeating levels not supported");
+        }
```
[the remark this answers is not in the document]
Comment (rcaudy) [reply extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java:471]: I understand now that this is dead code. We should delete it, and make a separate issue around optimizing based on the ideas here.

[the remark this answers is not in the document]
Comment (malhotrashivam) [reply extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java]: Created the issue #5512

**Review by rcaudy — APPROVED** [review]

**Pull request merged**